src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -221,6 +222,128 @@ def cli_cosmosdb_list(client, resource_group_name=None):
+            Location(location_name=loc.location_name, failover_priority=loc.failover_priority))
+
+    params = DatabaseAccountCreateUpdateParameters(
+        location=existing.location,
+        locations=locations,
+        tags=existing.tags,
+        kind=existing.kind,
+        consistency_policy=existing.consistency_policy,
+        ip_range_filter=existing.ip_range_filter,
+        enable_automatic_failover=existing.enable_automatic_failover,
+        capabilities=existing.capabilities,
+        is_virtual_network_filter_enabled=True,
+        virtual_network_rules=virtual_network_rules,
+        enable_multiple_write_locations=existing.enable_multiple_write_locations)
+
+    async_docdb_create = client.create_or_update(resource_group_name, account_name, params)
+    docdb_account = async_docdb_create.result()
+    docdb_account = client.get(resource_group_name, account_name)  # Workaround
+    return docdb_account
+
+
+def cli_cosmosdb_network_rule_remove(cmd,
+                                     client,
+                                     resource_group_name,
+                                     account_name,
+                                     subnet,
+                                     virtual_network=None):
+    """ Adds a virtual network rule to an existing Cosmos DB database account """
+    subnet = _get_virtual_network_id(cmd, resource_group_name, subnet, virtual_network)
+    existing = client.get(resource_group_name, account_name)
+
+    virtual_network_rules = []
+    rule_removed = False
+    for rule in existing.virtual_network_rules:
+        if rule.id != subnet:
+            virtual_network_rules.append(
+                VirtualNetworkRule(id=rule.id,
+                                   ignore_missing_vnet_service_endpoint=rule.ignore_missing_vnet_service_endpoint))
+        else:
+            rule_removed = True
+    if not rule_removed:
+        raise CLIError("This rule does not exist for the Cosmos DB account")
+
+    locations = []
+    for loc in existing.read_locations:
+        locations.append(
+            Location(location_name=loc.location_name, failover_priority=loc.failover_priority))
+
+    params = DatabaseAccountCreateUpdateParameters(
+        location=existing.location,
+        locations=locations,
+        tags=existing.tags,
+        kind=existing.kind,
+        consistency_policy=existing.consistency_policy,
+        ip_range_filter=existing.ip_range_filter,
+        enable_automatic_failover=existing.enable_automatic_failover,
+        capabilities=existing.capabilities,
+        is_virtual_network_filter_enabled=True,
+        virtual_network_rules=virtual_network_rules,
+        enable_multiple_write_locations=existing.enable_multiple_write_locations)
+
+    async_docdb_create = client.create_or_update(resource_group_name, account_name, params)
+    docdb_account = async_docdb_create.result()
+    docdb_account = client.get(resource_group_name, account_name)  # Workaround
```

Comment:
Your PUT doesn't return the object? You *should* be able to just:

```Python
return client.create_or_update(resource_group_name, account_name, params)
```

Reply:
I looked through our handler, and the only thing we currently return when the operation completes is {status: "Succeeded"}, so for the time being, the GET is needed to retrieve the object.